src/components/DERP/log.jsx

```diff
@@ -1,37 +1,102 @@
-import React, {useEffect} from "react";
+import React, { useEffect, useState } from "react";
+import styled from "@emotion/styled";
+
+const HighlightedAddress = styled.mark`
+  background-color: #ffa0a0;
+  font-weight: bold;
+`;
+
+const addressesToRegexes = (addresses) => {
+  return addresses.map((address) => {
+    const addressWithoutOx = address.slice(2);
+    return new RegExp(`(0x)?${addressWithoutOx}`, "gi");
+  });
+};
+
+function DERPLog({ log, addresses }) {
+  const [newLog, setNewLog] = useState([]);
+  const [regexes, setRegexes] = useState([]);
+
+  useEffect(() => {
+    setNewLog(JSON.parse(JSON.stringify(log)));
+  }, [log]);
+
+  useEffect(() => {
+    setRegexes(addressesToRegexes(addresses));
+  }, [addresses]);
+
+  const highlightParams = (params) => {
+    // make params a string and format it for <pre> tag.
+    let stringifiedParams = JSON.stringify(params, null, 2);
+
+    let newParams = [];
+    let index = 0;
+
+    regexes.forEach((regex) => {
+      const matches = [...stringifiedParams.matchAll(regex)];
+      if (matches.length === 0) {
+        return;
+      }
+
+      matches.forEach((match) => {
+        // index of the first match in the string.
+        const matchIndex = match.index;
+
+        // get the non-matching part of the string.
+        const nonMatchingPart = stringifiedParams.slice(index, matchIndex);
+
+        // get the matching part of the string (wallet address).
+        const matchPart = match.at(0);
+
+        if (nonMatchingPart) {
+          newParams = [...newParams, <span>{nonMatchingPart}</span>];
+        }
+
+        // newParams will be an array with strings and <HighlightedAddress> elements.
+        newParams = [...newParams, <HighlightedAddress>{matchPart}</HighlightedAddress>];
+
+        // sum of the first index match and address match length (40|42).
```

Comment:
what does the `(40|42)` mean?

Reply:
length of the address as it can be 40 (without 0x) or 42 (with 0x) but doesn't add anything to the comment, will remove it.